tests/unit/test_metadata.py

```diff
@@ -0,0 +1,125 @@
+        metadata.get(path=tmp_path)
+
+    assert metadata.METADATA_FILENAME in str(exc_info.value).lower()
+
+
+@pytest.mark.parametrize(
+    "metadata_yaml_content, expected_error_msg_contents",
+    [
+        pytest.param("", ("empty", metadata.METADATA_FILENAME), id="malformed"),
+        pytest.param(
+            "malformed: yaml:", ("malformed", metadata.METADATA_FILENAME), id="malformed"
+        ),
+        pytest.param("value 1", ("not", "mapping", metadata.METADATA_FILENAME), id="not dict"),
+    ],
+)
+def test_get_yaml_malformed(
+    metadata_yaml_content: str, expected_error_msg_contents: tuple[str, ...], tmp_path: Path
+):
+    """
+    arrange: given directory with metadata.yaml that is malformed
+    act: when get is called with the directory
+    assert: then InputError is raised.
+    """
+    create_metadata_yaml(content=metadata_yaml_content, path=tmp_path)
+
+    with pytest.raises(exceptions.InputError) as exc_info:
+        metadata.get(path=tmp_path)
+
+    assert_substrings_in_string(expected_error_msg_contents, str(exc_info.value).lower())
+
+
+# pylint currently does not understand walrus operator perfectly yet
+# pylint: disable=unused-variable,undefined-variable
+@pytest.mark.parametrize(
+    "metadata_yaml_content, expected_metadata_name, expected_metadata_docs",
+    [
+        pytest.param(
+            f"{metadata.METADATA_NAME_KEY}: {(name := 'name')}",
+            name,
+            None,
+            id="name only",
+        ),
+        pytest.param(
+            f"{metadata.METADATA_NAME_KEY}: {name}\n"
+            f"{metadata.METADATA_DOCS_KEY}: "
+            f"{(docs := 'https://discourse.charmhub.io/t/index/1')}",
+            name,
+            docs,
+            id="name and docs",
+        ),
+    ],
+)
+def test_get(
+    metadata_yaml_content: str,
+    expected_metadata_name: str,
+    expected_metadata_docs: str | None,
+    tmp_path: Path,
+):
+    """
+    arrange: given directory with metadata.yaml with valid metadata yaml
+    act: when get is called with the directory
+    assert: then file contents are returned as a dictionary.
+    """
+    create_metadata_yaml(
+        content=metadata_yaml_content,
+        path=tmp_path,
+    )
+
+    data = metadata.get(path=tmp_path)
+
+    assert data == types_.Metadata(name=expected_metadata_name, docs=expected_metadata_docs)
+
+
+@pytest.mark.parametrize(
+    "metadata_yaml_content, expected_error_msg_contents",
+    [
+        pytest.param("key: value", ("could not find", metadata.METADATA_NAME_KEY)),
+        pytest.param(
+            f"{metadata.METADATA_NAME_KEY}: 1",
+            (
+                "invalid value for",
+                metadata.METADATA_NAME_KEY,
+            ),
+        ),
+        pytest.param(
+            f"{metadata.METADATA_NAME_KEY}: name\n{metadata.METADATA_DOCS_KEY}: 1",
+            ("invalid value for", metadata.METADATA_DOCS_KEY),
+        ),
+    ],
```

Comment:
Should we also check for an empty `docs` key? That likely seems a mistake and will cause an error later anyway when the code tries to retrieve the value

Reply:
Done! Good point!